### ISSUE TITLE
Clarification Regarding Extension Field

### ISSUE THREAD

**Issue by calvinchai** [NeuroJSON/jnifti#9]
I came through the header specification and noticed that there is a [NIFTIExtension field](https://github.com/NeuroJSON/jnifti/blob/master/JNIfTI_specification.md#niftiextension:~:text=%7D%3B-,struct,%7D,-Notations%20from%20the) in the header table. If I understand correctly, the extension contents are store in the separated `NIFTIExtension` container, and the [section](https://github.com/NeuroJSON/jnifti/blob/master/JNIfTI_specification.md#niftiextension) talks about it. I assume the field in the JSON header is solely for the [4 bytes after Nifti header](https://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1.h#:~:text=After%20the%20end%20of%20the%20348%20byte%20header%20(e.g.%2C%20after%20the%20magic%20field)%2C%0A%20%20%20the%20next%204%20bytes%20are%20a%20char%20array%20field%20named%20%22extension%22.%20By%20default%2C%0A%20%20%20all%204%20bytes%20of%20this%20array%20should%20be%20set%20to%20zero.) and display them as a int array. Could you please clarify that my understand is correct? If so, is this field optional or mandatory, and what does the `<...>` part mean after this field.

**Comment by fangq** [comment]
@calvinchai, `NIFTIExtension` is optional, as [clarified here](https://github.com/NeuroJSON/jnifti/blob/v0.6/JNIfTI_specification.md?plain=1#L192-L193).

`<...>` is optional array or object elements, as [noted here](https://github.com/NeuroJSON/jnifti/blob/v0.6/JNIfTI_specification.md?plain=1#L287).

Because jnifti is aimed to be 100% backward compatible to nifti-1/2, so it uses this optional field to store extension data, which appears in some nifti files.

**Comment by calvinchai** [comment]
Thanks for the answer. In the link you provided, the [NIFTIExtension](https://github.com/NeuroJSON/jnifti/blob/master/JNIfTI_specification.md#niftiextension:~:text=%7D%3B-,struct,%7D,-Notations%20from%20the) field in JSON header became [NIIExtender](https://github.com/NeuroJSON/jnifti/blob/v0.6/JNIfTI_specification.md#:~:text=%22-,NIIExtender,-%22%3A%20%5B%3Ci%3E%2C%3Ci%3E%2C%3Ci), which is the one I want to ask about. Is this field optional or mandatory? Also which version (master or v0.6 tag) should I follow?

**Comment by fangq** [comment]
> I assume the field in the JSON header is solely for the [4 bytes after Nifti header](https://nifti.nimh.nih.gov/pub/dist/src/niftilib/nifti1.h#:~:text=After%20the%20end%20of%20the%20348%20byte%20header%20(e.g.%2C%20after%20the%20magic%20field)%2C%0A%20%20%20the%20next%204%20bytes%20are%20a%20char%20array%20field%20named%20%22extension%22.%20By%20default%2C%0A%20%20%20all%204%20bytes%20of%20this%20array%20should%20be%20set%20to%20zero.) and display them as a int array. Could you please clarify that my understand is correct? If so, is this field optional or mandatory

that's correct - the `NIFTIHeader.NIIExtender` subfield is only used to store the 349-352 bytes (total length of 4)  in the nifti1 header, or byte 541-544 in nifti-2 header. it is optional if the first byte is 0.

> what does the <...> part mean after this field

if you were referring to the `<...>` notation in [this line](https://github.com/NeuroJSON/jnifti/blob/v0.6/JNIfTI_specification.md?plain=1#L278), it means `NIFTIHeader` allows additional user-defined subfields. The intent was to allow wrapping other volume file headers, such as MGZ, that can not fit in NIFTI to be also stored in this data structure as extended fields. Maybe something like `NIFTIHeader.MGZHeader`, but I haven't get chance to read on what else should be carried here.

Also, please checkout the following sections of the spec
https://github.com/NeuroJSON/jnifti/blob/v0.6/JNIfTI_specification.md?plain=1#L291-L301
https://github.com/NeuroJSON/jnifti/blob/v0.6/JNIfTI_specification.md?plain=1#L195-L200

basically, the only item that is mandatory is `NIFTIData`.

**Comment by calvinchai** [comment]
Got it. Thanks.
